src/features/DebugSession.ts

```diff
@@ -42,6 +43,80 @@ export class DebugSessionFeature implements IFeature, DebugConfigurationProvider
         }));
     }
 
+    public async provideDebugConfigurations(
+        folder: WorkspaceFolder | undefined,
+        token?: CancellationToken): Promise<DebugConfiguration[]> {
+
+        const launchCurrentFileLabel  = "Launch Current File";
+        const launchScriptLabel       = "Launch Script";
+        const interactiveSessionLabel = "Interactive Session";
+
+        const debugConfigPickItems = [
+            {
+                label: launchCurrentFileLabel,
+                description: "Launch and debug the file in the currently active editor window",
+            },
+            {
+                label: launchScriptLabel,
+                description: "Launch and debug the specified file or command",
+            },
+            {
+                label: interactiveSessionLabel,
+                description: "Debug commands executed from the Integrated Console",
+            },
+            {
+                label: "Attach",
+                description: "Attach the debugger to a running PowerShell Host Process",
+            },
+        ];
+
+        const launchSelection =
```

Comment:
Still can't get the index of the selection, huh?

Reply:
Nah, I think you get back the selected item object.  I could add an index/id to the item objects and then test against that.  Worth it?

Reply:
Can you try it and let us know? I'm also curious for other situations as well.

Reply:
I looked at the API, it returns `Thenable<T | undefined>` where `T` is the type of the items array provided as input:

![image](https://user-images.githubusercontent.com/5177512/61649799-635e9100-ac6f-11e9-82b0-bf4548046ebc.png)

Reply:
Ah ok so in theory if our type had an index property, we could take advantage of it